components/Common/MetaBar/index.tsx

```diff
@@ -0,0 +1,69 @@
+import type { Heading } from '@vcarl/remark-headings';
+import { Fragment, useMemo } from 'react';
+import type { FC } from 'react';
+import { FormattedMessage } from 'react-intl';
+
+import LocalizedLink from '@/components/LocalizedLink';
+
+import styles from './index.module.css';
+
+type MetaBarProps = {
+  items: Record<string, React.ReactNode>;
+  headings?: {
+    items: Heading[];
+    depth?: number;
+  };
+};
+
+// The default depth of headings to display in the table of contents.
+const DEFAULT_HEADING_DEPTH: number = 2;
+
+const MetaBar: FC<MetaBarProps> = ({
+  items,
+  headings = {
+    items: [],
```

Comment:
I would rather set the default depth here. Instead of down there,

```suggestion
    items: [],
    depth: DEFAULT_HEADING_DEPTH
```

Reply:
> I would rather set the default depth here. Instead of down there,

The value in the object we set as default becomes `undefined` because the object is overwritten. We can solve this by deconstructing headings and setting them as default but IMO looks worse than this solution

Reply:
> The value in the object we set as default becomes undefined because the object is overwritten. We can solve this by deconstructing headings

What is overwritten (who overwrites what?)

I think instead of creating a default prop directly on the `props deconstruction` you could do:

```tsx
const { depth = DEFAULT_HEADING_DEPTH, items = [] } = headings;
```

Maybe?